src/ui/InteractiveBookShelf/index.tsx

```diff
@@ -1,39 +1,57 @@
 'use client';
 
 import { APIDefaultBook } from '@/types/book';
+import { APIBookshelfBookList } from '@/types/bookshelf';
 import InteractiveBook from '@/ui//InteractiveBook';
 import { Flex, useTheme } from '@chakra-ui/react';
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 
-type BookListTypes = {
-  bookList: APIDefaultBook[];
-};
+const BOOKSHELF_BOOK_LIMIT = 4;
 
-const InteractiveBookShelf = ({ bookList }: BookListTypes) => {
+const InteractiveBookShelf = ({
+  books,
+}: {
+  books: APIBookshelfBookList['books'] | APIDefaultBook[];
```

Comment:
ask;

```suggestion
  books: APIDefaultBook[];
```

로 바꿔도 될 것 같은데, 안되나요? 👀

Reply:
오 신기하게도 되네요? 타입스크립트 알다가도 모르겠따 😢